expression/builtin_time_vec.go

```diff
@@ -697,11 +697,33 @@ func (b *builtinGetFormatSig) getFormat(format, location string) string {
 }
 
 func (b *builtinLastDaySig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinLastDaySig) vecEvalTime(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	if err := b.args[0].VecEvalTime(b.ctx, input, result); err != nil {
+		return err
+	}
+	times := result.Times()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if times[i].InvalidZero() {
+			return handleInvalidTimeError(b.ctx,
+				types.ErrIncorrectDatetimeValue.GenWithStackByArgs(times[i].String()))
+		}
+		tm := times[i].Time
+		year, month := tm.Year(), tm.Month()
```

Comment:
A curiosity is that why this judgment is deleted?
https://github.com/pingcap/tidb/blob/6b9703bc5f8c77ce3350c8880958b44fef53b7b7/expression/builtin_time.go#L6536-L6538

Reply:
It was not deleted. It's the same as `times[i].InvalidZero()`

Reply:
That's great, could you also modify the original implementation `builtinLastDaySig.evalTime`.

Reply:
Done